programs/mmm/src/instructions/vanilla/sol_fulfill_buy.rs

```diff
@@ -175,40 +177,24 @@ pub fn handler<'info>(
 
     // check creator_accounts and verify the remaining accounts
     let creator_accounts = if pool.using_shared_escrow() {
-        // check the remaining accounts at position 0 and 1
-        // 0 has to be the m2_program
-        // 1 has to be the shared_escrow_account pda of the m2_program
-        if ctx.remaining_accounts.len() < 2 {
-            return Err(MMMErrorCode::InvalidRemainingAccounts.into());
-        }
-        if *ctx.remaining_accounts[0].key != M2_PROGRAM {
-            return Err(MMMErrorCode::InvalidRemainingAccounts.into());
-        }
-        let shared_escrow_account = &ctx.remaining_accounts[1];
-        let pda_program =
-            check_buyside_sol_escrow_account(&shared_escrow_account.key(), &pool_key, &pool.owner)?;
-        if pda_program != M2_PROGRAM {
-            return Err(MMMErrorCode::InvalidRemainingAccounts.into());
-        }
+        check_remaining_accounts_for_m2(remaining_accounts, &pool_key, &pool.owner.key())?;
 
-        // TODO
-        // Change to the correct amount to be withdrawn from shared escrow
-        // into the buyside_sol_escrow_account
         let amount: u64 = (total_price as i64 + maker_fee) as u64;
         withdraw_m2(
             pool,
             ctx.bumps.pool,
             buyside_sol_escrow_account,
-            shared_escrow_account,
+            index_ra!(remaining_accounts, 1),
             system_program,
-            &ctx.remaining_accounts[0],
+            index_ra!(remaining_accounts, 0),
             pool.owner,
             amount,
         )?;
+        pool.shared_escrow_cap = Some(pool.shared_escrow_cap.unwrap().checked_sub(amount).unwrap());
```

Comment:
for double sided pool, only subtraction will make the pool not self-sustainable? Should we add back the cap when a solfulfill sell happen?

Reply:
we should discuss the double sided pool later, there're some implications on how shared escrow balance might increase and that will negatively impact the UX, like accidentally reactivate the pools.

and yes, we can have a new PR for double sided pool with shared escrow